### PR TITLE
added memo with tests and verify_signature bool 

### PR DESCRIPTION
**Pull request by crypt0miester**
* Added memo program with tests.
* Added memo docs 
* Added verify signatures for transaction.serialize() since sometimes the frontend is the fee payer.
This uses the same functionality as [web3.js](https://github.com/solana-labs/solana/blob/363d8d1539e8cebc2d0921024d9edc35a38a4e34/web3.js/src/transaction.ts#L706) 
* Added unit test to make sure that the verify signature=False would work.

### CODE REVIEW

File: src/solana/transaction.py [hunk 1]
```diff
@@ -333,10 +335,14 @@ def verify_signatures(self) -> bool:
             return False
         return True
 
-    def serialize(self) -> bytes:
+    def serialize(self, verify_signatures: bool = True) -> bytes:
```
Comment (michaelhly) [inline src/solana/transaction.py:338]: Why is this change necessary?

Comment (crypt0miester) [reply src/solana/transaction.py:338]: * Added verify signatures for transaction.serialize() since sometimes the frontend is the fee payer.
This uses the same functionality as [web3.js](https://github.com/solana-labs/solana/blob/363d8d1539e8cebc2d0921024d9edc35a38a4e34/web3.js/src/transaction.ts#L706)

^^ referenced in OP

so basically if the fee payer has not signed the txn, serialize() would fail if verify signatures is True. 

This is necessary for platforms that use the same design which is as follows: 
* backend creates txn and signs it (verify_signatures = False)
* send txn to frontend
* fee payer signs with (verifySignature = True) in web3.js

they even have requireAllSignature which hasn't been implemented in solana-py.

**Review by michaelhly — APPROVED** [review]

**Comment by michaelhly** [comment]
@kevinheavey everything looks find to you?

**Comment by kevinheavey** [comment]
Yeah lgtm
